File: analisis/src/Tarea5.java
import java.util.Random;
import java.util.Scanner;
import java.util.Arrays;

public class Tarea5 {
    // La operación dominante es la comparación dentro de los bucles anidados.
    // Complejidad O(n^2).
    public static void bubbleSort(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            for (int j = 0; j < arr.length - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter number of elements: ");
        int n = sc.nextInt();
        sc.close(); // Se cierra el scanner para evitar fugas de recursos

        int[] data = new int[n];
        Random rand = new Random();

        // Llenar el arreglo con datos aleatorios
        for (int i = 0; i < n; i++) {
            data[i] = rand.nextInt(1000);
        }

        // Medición del tiempo de ejecución [cite: 55]
        long startTime = System.nanoTime(); // Iniciar medición [cite: 55]
        bubbleSort(data); // Ejecutar el algoritmo
        long endTime = System.nanoTime(); // Finalizar medición [cite: 55]
        long duration = (endTime - startTime) / 1000000; // Duración en milisegundos

        System.out.println("Sorted array (first 10 elements): " + Arrays.toString(Arrays.copyOf(data, Math.min(n, 10))));
        System.out.println("Execution time: " + duration + " ms");
    }
}

File: analisis/src/DANIEL/tarea5.py
import random
import time

def bubble_sort(arr):
    for i in range(len(arr) - 1):
        for j in range(len(arr) - i - 1):
            if arr[j] > arr[j + 1]:
                temp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = temp

def main():
    n = int(input("Enter number of elements: "))
    data = [random.randint(0, 999) for _ in range(n)]

    start_time = time.time()
    bubble_sort(data)
    end_time = time.time()

    duration = (end_time - start_time) * 1000  # convertir a milisegundos

    print("Sorted array (first 10 elements):", data[:min(n, 10)])
    print(f"Execution time: {duration:.2f} ms")

if __name__ == "__main__":
    main()
